Content.Shared/Alert/AlertsSystem.cs

```diff
@@ -176,6 +180,24 @@ public override void Initialize()
         LoadPrototypes();
     }
 
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<AlertsComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            foreach (var (alertKey,  alertState) in comp.Alerts)
+            {
+                if (alertState.AutoRemove is null || alertState.AutoRemove >= _timing.CurTime)
+                    continue;
+
+                comp.Alerts.Remove(alertKey);
+                Dirty(uid, comp);
+            }
+        }
+    }
+
```

Comment:
This looks pretty expensive if you're iterating every single mob and then some + all of their alerts dictionaries compared to just using another component.

Reply:
I am now making essentially a copy-set of these particular alertcomponents with just the relevant alerts, for easier searching. Is this acceptable?